mscore/plugin/api/elements.cpp

```diff
@@ -69,6 +69,42 @@ void Note::setTpc(int val)
             set(Pid::TPC2, val);
       }
 
+
+//---------------------------------------------------------
+//   Chord::add
+//---------------------------------------------------------
+
+void Chord::add(Ms::PluginAPI::Element* wrapped)
+      {
+      Ms::Element* s = wrapped->element();
+      if (s)
+            {
+            // Score now owns the object.
+            wrapped->setOwnership(Ownership::SCORE);
+            // Provide parentage for element.
+            s->setParent(chord());
+            // Create undo op and add the element.
+            chord()->score()->undoAddElement(s);
+            // If a note, ensure the element has proper Tpc values. (Will crash otherwise)
+            if (s->type() == ElementType::NOTE)
+                  toNote(s)->setTpcFromPitch();
+            }
+      }
+
+//---------------------------------------------------------
+//   Chord::remove
+//---------------------------------------------------------
+
+void Chord::remove(Ms::PluginAPI::Element* wrapped)
+      {
+      Ms::Element* s = wrapped->element();
+      if (chord()->notes().size() <= 1 && s->type() == ElementType::NOTE)
```

Comment:
As this function can be called with any chord and any element, it might be good to ensure first that `s->parent() == chord()`, otherwise the chord size check and the whole removal operation may do something unintended.

Reply:
Good point. It also encourages using the deleteElement() as well.
-Dale

Reply:
Done.